test/Get-GitDirectory.Tests.ps1

```diff
@@ -0,0 +1,69 @@
+. $PSScriptRoot\Shared.ps1
+
+Describe 'Get-GitDiretory Tests' {
+    Context "Test normal repository" {
+        BeforeAll {
+            $origPath = Get-Location
+        }
+        AfterAll {
+            Set-Location $origPath
+        }
+
+        It 'Returns $null for not a Git repo' {
+            Set-Location $env:windir
+            Get-GitDirectory | Should BeNullOrEmpty
+        }
+        It 'Returns $null for not a filesystem path' {
+            Set-Location Cert:\CurrentUser
+            Get-GitDirectory | Should BeNullOrEmpty
+        }
+        It 'Returns correct path when in the root of repo' {
+            $repoRoot = (Resolve-Path $PSScriptRoot\..).Path
+            Set-Location $repoRoot
+            Get-GitDirectory | Should BeExactly (MakeNativePath $repoRoot\.git)
+        }
+        It 'Returns correct path when under a child folder of the root of repo' {
+            $repoRoot = (Resolve-Path $PSScriptRoot\..).Path
+            Set-Location $PSScriptRoot
+            MakeNativePath (Get-GitDirectory) | Should BeExactly (Join-Path $repoRoot .git)

```

Comment:
Because git rev-parse returns paths like `C:/users/keith/...` even on Windows.  Hmm, maybe we should make Get-GitDirectory return a platform appropriate path.  RE 1 & 2, I do test when path is root or posh-git repo and again when path is root\test.  Remember in the test, $PSScriptRoot is the test dir.  And I'm not too concerned with the Pester tests not being able to run on PS v2. Sound reasonable?